bookstore/publish.py

```diff
@@ -0,0 +1,80 @@
+import json
+
+import aiobotocore
+
+from notebook.base.handlers import APIHandler
+from notebook.base.handlers import path_regex
+from notebook.utils import url_path_join
+from tornado import web
+
+from .s3_paths import s3_path
+from .s3_paths import s3_key
+from .s3_paths import s3_display_path
+from .bookstore_config import BookstoreSettings
+
+
+class BookstorePublishHandler(APIHandler):
+    """Publish a notebook to the publish path"""
+
+    def initialize(self):
+        """Initialize a helper to get bookstore settings and session information quickly"""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+        self.session = aiobotocore.get_session()
+
+    @web.authenticated
+    async def put(self, path=''):
```

Comment:
Why have the default if it will 400 anyway? Is this needed for the @web decorator?